File: core/src/main/java/ml/dmlc/mxnet/LibInfo.java
package ml.dmlc.mxnet;

import ml.dmlc.mxnet.Base.*;

class LibInfo {
  public native int mxNDArrayFree(NDArrayHandle handle);
  public native String mxGetLastError();
  public native int mxNDArrayCreateNone(NDArrayHandle out);
}

File: core/src/main/scala/ml/dmlc/mxnet/LibInfo.scala
package ml.dmlc.mxnet

import ml.dmlc.mxnet.Base._

import scala.collection.mutable.ListBuffer

class LibInfo {
  @native def mxNDArrayFree(handle: NDArrayHandle): Int
  @native def mxGetLastError(): String
  @native def mxNDArrayCreateNone(out: NDArrayHandle): Int
  @native def mxNDArrayCreate(shape: Array[Int], ndim: Int, devType: Int,
                              devId: Int, delayAlloc: Int, out: NDArrayHandle): Int
  @native def mxNDArrayWaitAll(): Int
  @native def mxListFunctions(functions: ListBuffer[FunctionHandle]): Int
}
